tests/PHPStan/Analyser/data/generics.php

```diff
@@ -344,3 +344,54 @@ public function g() {
 		assertType('T (class PHPStan\Generics\FunctionsAssertType\C, argument)', $a->g());
 	}
 }
+
+/**
+ * class A
+ *
+ * @template T
+ */
+class A {
+	/**
+	 * A::__construct()
+	 *
+	 * @param T $a
+	 */
+	public function __construct($a) {
+	}
+}
+
+/**
+ * class AOfDateTime
+ *
+ * @extends A<\DateTime>
+ */
+class AOfDateTime extends A {
+}
+
+/**
+ * class B
+ *
+ * @template T
+ *
+ * @extends A<T>
+ */
+class B extends A {
+	/**
+	 * B::__construct()
+	 *
+	 * @param T $a
+	 */
+	public function __construc($a) {
```

Comment:
Typo in the constructor. How it's possible that this works? :)

Reply:
Good catch ! /o\

Inference uses the parent constructor instead, and this works by accident. We should prevent new-expression type inference when the instantiated class does not have a constructor.

Reply:
Nice, we figured out a use case to cover thanks to a typo in a test :D